src/fetcher.ts

```diff
@@ -61,22 +62,29 @@ export abstract class Fetcher {
    * Fetches information about a pair and constructs a pair from the given two tokens.
    * @param tokenA first token
    * @param tokenB second token
-   * @param provider the provider to use to fetch the data
+   * @param connex the provider to use to fetch the data
    */
   public static async fetchPairData(
     tokenA: Token,
     tokenB: Token,
-    connex?: any
+    connex: any
   ): Promise<Pair> {
     invariant(tokenA.chainId === tokenB.chainId, 'CHAIN_ID')
     const pairAddress = Pair.getAddress(tokenA, tokenB)
 
-    const getReservesABI = find(IUniswapV2Pair.abi, { name: 'getReserves' });
+    const getReservesABI = find(IVexchangeV2Pair.abi, { name: 'getReserves' });
     const getReservesMethod = connex.thor.account(pairAddress).method(getReservesABI);
-  
-    const reserves = await getReservesMethod.call().then((data: any) => data.decoded)
-    const { reserve0, reserve1 } = reserves;
+
+    const reserves = (await getReservesMethod.call()).decoded
+    const { reserve0, reserve1 } = reserves
     const balances = tokenA.sortsBefore(tokenB) ? [reserve0, reserve1] : [reserve1, reserve0]
-    return new Pair(new TokenAmount(tokenA, balances[0]), new TokenAmount(tokenB, balances[1]))
+
+    const swapFeeABI = find(IVexchangeV2Pair.abi, { name: 'swapFee' })
+    const method = connex.thor.account(pairAddress).method(swapFeeABI)
+    const swapFee = JSBI.BigInt((await method.call()).decoded['0'])
```

Comment:
I'd prefer to have the await on a different line here. Its doing multiple things in this one line

Reply:
fixed, please see if it's okay now